stock_partner_delivery_window/models/stock_picking.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from datetime import timedelta
+
+from odoo import _, api, models
+from odoo.tools.misc import format_datetime
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    @api.onchange("scheduled_date")
+    def _onchange_scheduled_date(self):
```

Comment:
Why only check in the onchange? this seems to me a little bit weak, you can write any scheduled_date with any other action. Why don't you extend '_compute_scheduled_date' instead or use a api.constraint?

Reply:
IIRC the idea was to still allow overriding and get the warning in such case. 

@jgrandguillaume can you confirm we don't want to restrict this?